java/Cherepnin_Oleh/src/com/zoolatech/lecture3/tasks/_3/Triangle.java

```diff
@@ -0,0 +1,67 @@
+package com.zoolatech.lecture3.tasks._3;
+
+import java.util.Objects;
+
+public final class Triangle implements Shape {
+
+    private int aLength;
+    private int bLength;
+    private int cLength;
+
+    public Triangle(int aLength) {
+        this(aLength, aLength, aLength);
+    }
+
+    public Triangle(int aLength, int bLength, int cLength) {
+        this.aLength = aLength;
+        this.bLength = bLength;
+        this.cLength = cLength;
+    }
+
+    @Override
+    public double findPerimeter() {
+        // P = a + b + c
+        if (checkTriangle()) {
+            return aLength + bLength + cLength;
+        }
+        throw new ArithmeticException("Triangle does not exist");
+    }
+
+    private boolean checkTriangle() {
```

Comment:
Probably, this method should be called during the object construction. Also, it's better to use `IllegalArgumentException` instead of `ArithmeticException`